File: es3/es3.go
package main

import (
	"fmt"
	"os"
)

func main() {
	input, err := os.Open("input.txt")
	if err != nil {
		panic(err)
	}
	defer input.Close()

	fabric := [1000][1000]int{}
	for {
		var id, x, y, w, h int

		_, err := fmt.Fscanf(input, "#%d @ %d,%d: %dx%d\n", &id, &x, &y, &w, &h)
		if err != nil {
			break
		}

		for i := y; i < y+h; i++ {
			for j := x; j < x+w; j++ {
				fabric[i][j]++
			}
		}
	}

	sqinches := 0
	for i := 0; i < 1000; i++ {
		for j := 0; j < 1000; j++ {
			if fabric[i][j] > 1 {
				sqinches++
			}
		}
	}
	fmt.Println("Square inches:", sqinches)
}

File: es3/es3.py
#!/usr/bin/env python3

import re

fabric = [[0 for x in range(1000)] for y in range(1000)]
pattern = re.compile(r"^#(\d+) @ (\d+),(\d+): (\d+)x(\d+)$")
with open("input.txt", "r") as f:
    for line in f:
        m = pattern.match(line)
        if m:
            x1 = int(m.group(2))
            y1 = int(m.group(3))
            w = int(m.group(4))
            h = int(m.group(5))
            for y in range(y1, y1 + h):
                for x in range(x1, x1 + w):
                    fabric[y][x] += 1

square_inches = 0
for y in range(0, 1000):
    for x in range(0, 1000):
        if fabric[y][x] > 1:
            square_inches += 1

print("Square inches:", square_inches)
